src/rt/dmain2.d

```diff
@@ -321,7 +410,7 @@ private alias extern(C) int function(char[][] args) MainFunc;
  * runs embedded unittests and then runs the given D main() function,
  * optionally catching and printing any unhandled exceptions.
  */
-extern (C) int _d_run_main(int argc, char** argv, MainFunc mainFunc)
+export extern (C) int _d_run_main(int argc, char** argv, MainFunc mainFunc)
```

Comment:
AFAICT `export` produces exported symbols on every executable, too. I would recommend just using a def file in case you actually need any of these symbols being exported.

Reply:
oh yeah i meant to remove that before pushing it here but missed a spot.

I was also experimenting with actually making druntime.dll and sprinkled `export` in parts around the runtime to get it to load. I actually got it sort of working but virtual functions always jumped to 0xffffffffff - i figure that data must not have been indirected correctly by dmd then scaled back my ambitions to this PR.

(I think druntime.dll is the correct solution for my plugin problem but I ran out of time trying to get that path to work.)

Reply:
For proper shared runtime/library support, see https://dconf.org/2016/talks/thaut.html

Reply:
> oh yeah i meant to remove that before pushing it here but missed a spot.

I meant all other functions with changed export, too. For DLL support, I implemented an `-exportall` switch a couple of years ago. It does what's also done on linux, i.e. export all symbols.

Reply:
Yeah, I looked at Thaut's code too, it looks like it did most the work reasonably well so I might try to revive that fork and get it merged when my urgent todo list is a bit shorter.

When/if that gets in, I would then delete all this new code and the GC proxy stuff too I'm pretty sure is useless witht hat.

Reply:
> When/if that gets in, I would then delete all this new code and the GC proxy stuff too I'm pretty sure is useless witht hat.

I don't remember whether Benjamins' code handled the non-standard exception handling in Win64. I'm not familiar with the implementation, but it doesn't look like it would work out of the box by just patching up import relocations.